File: clients/node/examples/basic.ts
/**
 * 基本使用示例
 */

import { WingmanClient } from '@wingman/client';

async function main() {
  const client = new WingmanClient({
    host: 'localhost',
    port: 8080
  });

  try {
    // 连接
    await client.connect();
    console.log('已连接到服务器');

    // 获取屏幕尺寸
    const size = await client.screen.getSize();
    console.log(`屏幕尺寸: ${size.width}x${size.height}`);

    // 获取鼠标位置
    const pos = await client.input.getMousePosition();
    console.log(`鼠标位置: (${pos.x}, ${pos.y})`);

    // 获取系统信息
    const cpu = await client.system.getCpuInfo();
    console.log(`CPU: ${cpu.brand} (${cpu.cores} 核心)`);

    const mem = await client.system.getMemoryInfo();
    console.log(`内存: ${(mem.total / 1024 / 1024 / 1024).toFixed(2)} GB`);

  } finally {
    await client.disconnect();
    console.log('已断开连接');
  }
}

main().catch(console.error);

File: clients/python/examples/basic.py
"""
Wingman Python Client 基本使用示例
"""

from wingman import WingmanClient

# 使用上下文管理器（推荐）
with WingmanClient(host='localhost', port=8080) as client:
    # 获取屏幕尺寸
    screen = client.screen.get_size()
    print(f"屏幕尺寸: {screen['width']}x{screen['height']}")

    # 获取鼠标位置
    mouse = client.input.get_mouse_position()
    print(f"鼠标位置: ({mouse['x']}, {mouse['y']})")

    # 获取系统信息
    cpu = client.system.get_cpu_info()
    print(f"CPU: {cpu['brand']} ({cpu['cores']} 核心)")

    mem = client.system.get_memory_info()
    print(f"内存: {mem['total'] / 1024 / 1024 / 1024:.2f} GB")

print("\n示例完成！")
